frontend/workflows/projectSelector/src/selector-reducer.tsx

```diff
@@ -97,13 +151,32 @@ const selectorReducer = (state: State, action: Action): State => {
       return newState;
     }
     case "REMOVE_PROJECTS": {
-      // TODO: also remove any upstreams or downstreams related (only) to the project.
-      // if group == Groups.PROJECT, hide exclusive downstream upstreams
-      //
-      return {
+      // remove the projects from their respective group
+      const newState = {
         ...state,
         [action.payload.group]: _.omit(state[action.payload.group], action.payload.projects),
       };
+
+      if (action.payload.group !== Group.PROJECTS) {
+        return newState;
+      }
+
+      // for the removed projects in Group.Projects we get their upstreams/dowstreams that are exclusive
+      // (i.e. not shared by other projects that weren't removed)
```

Comment:
unfortunately i think there's a bug here.

let's say we have

```
foo => a, b
bar => b, c
baz => x, y
```

and we call `remove(foo, bar)`

the result should be `[x, y]` but i think the current code would result in `[b, x, y]`

we could leave a TODO i guess. this complexity is an unfortunate result of my choice to put in multiple projects.

Reply:
ooh i see - in the func I check to see if the dependency map is a length of 1 but since I'm only checking it for 1 project at at time, the removed projects could be sharing that same dependency but it wouldn't be removed. i'll leave a TODO and followup